Harmony/Public/Patching/NativeDetourMethodPatcher.cs

```diff
@@ -122,12 +118,12 @@ public override DynamicMethodDefinition CopyOriginal() =>
 
 	private DynamicMethodDefinition GenerateAltEntryProxyCaller(string name)
 	{
-		var dmd = _staticSignature.CreateDmd(name);
+		var dmd = _signature.CreateDmd(name);
 
 		var il = new ILContext(dmd.Definition);
 		var c = new ILCursor(il);
 
-		for (var i = 0; i < _staticSignature.ParameterCount; i++)
+		for (var i = 0; i < _signature.ParameterCount; i++)
 			c.EmitLdarg(i);
```

Comment:
What about the instance parameter?

Reply:
thats the thing, ParameterCount is already properly set through through MethodSignature, unless you set "ignoreThis=true", which it was previously.

Reply:
```csharp
        public MethodSignature(MethodBase method, bool ignoreThis)
        {
            ReturnType = (method as MethodInfo)?.ReturnType ?? typeof(void);

            var thisCount = ignoreThis || method.IsStatic ? 0 : 1;

            var methParams = method.GetParameters();
            parameters = new Type[methParams.Length + thisCount];

            for (var i = thisCount; i < parameters.Length; i++)
            {
                parameters[i] = methParams[i - thisCount].ParameterType;
            }

            if (!ignoreThis && !method.IsStatic)
            {
                parameters[0] = method.GetThisParamType();
            }
        }
```